### DISCUSSION TITLE
Allow renovate-config-validator to Throw an Error when Migration is Necessary

### DISCUSSION THREAD

**Discussion by bachmacintosh** [renovatebot/renovate#22016] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

I noticed while updating my shared Renovate configs today that `stabilityDays` is deprecated in favor of `minimumReleaseAge`. The only reason I noticed this was because, despite my config validation check in the PR passing since there are no errors, I happened to check the logs anyway, and saw the raised warning. Had I not double-checked the logs, there's a chance that sometime in the future a bunch of repos using the shared configs would raise issues from Renovate in some fashion.

I check the code for the validator, and the only argument it takes is the configuration file path. Ideally there would be an option like `--treatMigrationsAsErrors` or something like it, that when passed in on the command line, [the check if the config needs migrating](https://github.com/renovatebot/renovate/blob/2112ea26461491fd7afb60b8b57011a134f3cbdf/lib/config-validator.ts#L24) could throw an error instead. This of course would be optional and default to `false` for backwards compatibility, but could also allow users to catch these deprecations and be proactive in updating their configs, avoiding potential errors down the road.

Thanks!

**Comment by HonkingGoose** [comment]
I think our plan is to enable configuration migration PRs by default, when the feature is ready for public use. You would get a "config migration PR" from Renovate, review it and then merge it.

- #16359 

I'll let the maintainers review your idea.

https://docs.renovatebot.com/configuration-options/#configmigration

**Comment by rarkins** [comment]
I think the flag could possibly be e.g. `--strict`

**Reply by HonkingGoose** [reply]
I think other CLI programs also often use `--strict` so we would match users's expectations there.

@rarkins Do you want an issue to track this feature?

**Reply by rarkins** [reply]
Yes, a "Add strict mode to config validator" feature request

**Reply by HonkingGoose** [reply]
Here's your issue:

- #22026
